### PR TITLE
Test epic

### LINKED ISSUE #62: extract_epic.py tests
## Description
- Write Unit tests for `extract_epic.py `

## Required Files
- `extract_epic.py`

## User Story
As a quality assurance engineer it is vital to test the functionality of the extract script to ensure that the script is functioning correctly and prevent any unexpected bugs.

### PR DESCRIPTION
**Pull request by shindym**
- Unit tests added to test extracting data from epic games
- conftest.py file added
-  extract_epic.py script updated to have extract process function to run whole extraction process.

Closes #62

### CODE REVIEW

File: pipeline/test_epic.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+"""This file contains tests for the extract_epic.py functions."""
+import datetime
+from extract_epic import get_games_data, get_game_details, epic_extract_process
+
+
+def test_pull_data_from_graphql(requests_mock, epic_game_post_response):
+    """tests successful post request"""
```
Comment (annalisev) [inline pipeline/test_epic.py:7]:
```suggestion
    """Tests successful post request."""
```

File: pipeline/test_epic.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    requests_mock.post('http://www.epic/test/graphql.com',
+                       json=epic_game_post_response)
+    result = get_games_data(config)
+
+    assert result == epic_game_post_response['data']['Catalog']['searchStore']['elements']
+
+
+def test_correct_game_details_pulled(epic_game_data_game_obj):
+    """tests successful extracting of data from json object."""
```
Comment (annalisev) [inline pipeline/test_epic.py:20]:
```suggestion
    """Tests successful extracting of data from json object."""
```

File: pipeline/test_epic.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    assert result == ['Vengeance of Mr. Peppermint',
+                      "Long ago, they killed his sister. Now, he will kill them.",
+                      19.99, 'Hack The Publisher', 'Freedom Games', datetime.datetime(
+                          2024, 4, 24, 16, 0), None, 3,
+                      ['Action', 'Action-Adventure', 'Single Player',
+                       'Windows', 'Indie', 'Mac OS'], [1, 2]]
+
+
+def test_no_new_games(requests_mock, epic_empty_post_request):
+    """Tests error handling of grabbing data"""
```
Comment (annalisev) [inline pipeline/test_epic.py:32]:
```suggestion
    """Tests error handling of grabbing data."""
```

File: pipeline/test_epic.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        'BASE_URL': 'http://www.epic/test/graphql.com'
+    }
+    requests_mock.post('http://www.epic/test/graphql.com',
+                       json=epic_empty_post_request)
+    result = epic_extract_process(config)
+    assert result == []
+
+
+def test_no_one_new_games(requests_mock, epic_game_post_response):
+    """Tests error handling of grabbing data"""
```
Comment (annalisev) [inline pipeline/test_epic.py:43]:
```suggestion
    """Tests error handling of grabbing data."""
```

**Review by annalisev — CHANGES_REQUESTED** [review]
Few small changes, nothing major- what is the test coverage of these tests?

**Comment by shindym** [comment]
> Few small changes, nothing major- what is the test coverage of these tests?

95% coverage

**Review by annalisev — APPROVED** [review]
Perfect!

File: pipeline/test_epic.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
+"""This file contains tests for the extract_epic.py functions."""
+import datetime
+
+from extract_epic import get_games_data, get_game_details, epic_extract_process
+
+
+def test_pull_data_from_graphql(requests_mock, epic_game_post_response):
+    """Tests successful post request."""
+
+    config = {
```
Comment (hhm970) [inline pipeline/test_epic.py:10]: `.env` file nicely mocked!

**Review by hhm970 — APPROVED** [review]
Tests looks sound and mock all impurities in each function well!

**Pull request merged**